server/src/main/java/org/opensearch/index/mapper/MappedFieldType.java

```diff
@@ -536,4 +536,12 @@ public Map<String, String> meta() {
     public TextSearchInfo getTextSearchInfo() {
         return textSearchInfo;
     }
+
+    public void encodePoint(Number value, byte[] point) {
+        throw new IllegalCallerException("Field [" + name() + "] of type [" + typeName() + "] does not support encoding points");
+    }
+
+    public int pointNumBytes() {
+        throw new IllegalCallerException("Field [" + name() + "] of type [" + typeName() + "] does not support encoding points");
+    }
```

Comment:
Why do you need both of these methods? Why couldn't `encodePoint` just allocate and return `byte[]`?

Also, we shouldn't add these to `MappedFieldType`, since they're not generally applicable.

What about adding an interface called `PointFieldType` that declares `encodePoint`? Each of these classes can implement `PointFieldType`, then you can do an `instanceOf` check in `buildRanges` to see if you can safely cast and invoke the method.

Reply:
Addressed the comment, thanks!